### ISSUE TITLE
[UI EPIC] Add notification centre to UI

### ISSUE THREAD

**Issue by gaktive** [rancher/dashboard#14007]
For management tracking purposes.

**Comment by yonasberhe23** [comment]
Manual validations required

**Comment by IsaSih** [comment]
I'm reopening the issue as the "Mark all as read" option is not accessible via keyboard navigation. Validated in 

[Rancher](https://github.com/rancher/rancher)	v2.12-659a3fe3ac0d6ff0654717dcbd3161391d1f5662-head
[Dashboard](https://github.com/rancher/dashboard)	master bdce41a
[Rancher CLI](https://github.com/rancher/cli)	v2.12.0-rc.1

https://github.com/user-attachments/assets/c94d65ce-f786-4a29-bcdb-18fe10ef0d3c

**Comment by nwmac** [comment]
@IsaSih It should be - when you open the notification center, you use up/down keys to cycle, so press down and 'Mark all as read' will gain focus. Press down/up to then cycle through each notification. You press enter/space on a notification to then be able to interact with the read/unread toggle for an individual notification and the action buttons (if present). You need to press ESCape to exit focus of the notification, to return so you can use up/down to cycle through the notifications and the 'Mark all as read' option

**Comment by nwmac** [comment]
@IsaSih All work is now merged, so this is ready to test once again

**Comment by IsaSih** [comment]
Validated in 

[Rancher](https://github.com/rancher/rancher)	v2.12-d1bb158637d9a02a85ee57e98b182b7c3bb6b423-head
[Dashboard](https://github.com/rancher/dashboard)	master 2e25bc8
[Rancher CLI](https://github.com/rancher/cli)	v2.12.0-rc.1

If validating the Notifications Center via the notifications-test UI extension, all the mock cases of the test extension itself work as expected, e.g., clicking "Add a test Growl" indeed displays a growl that is visible in the notifications center.

https://github.com/user-attachments/assets/d5622a4b-dc33-455a-9052-8df3ef60a043
https://github.com/user-attachments/assets/4aae30f4-18c8-41c4-9feb-25a372639bab

 However, by reproducing real use cases in Rancher, the same notifications of the testing extension should be triggered, and the notification center does not get updated, nor are the growls being displayed. 

For example, in the testing extension, a sample growl is "Virtual Clusters UI Extension loaded successfully", so I did install the Virtual Clusters UI extension, but the growl is not displayed, nor is the notification center updated, as it is by mocking this scenario with the notifications-test UI extension.

<img width="1901" height="925" alt="Image" src="https://github.com/user-attachments/assets/2df1bbfd-b4c0-433d-9b2e-dcbb413fd2bc" />

https://github.com/user-attachments/assets/f6877a56-4d05-4494-bd9e-b14a101feb47

Same for the progress bar notification when a cluster is provisioning. Mocking this case works in the test extension, but when provisioning a cluster in the Rancher dashboard, nothing is displayed in the notification center.

For real use cases in the Rancher dashboard, the only notification that is displayed is for the notification center is
"Welcome to Rancher v2.12" - release notes.

**Comment by richard-cox** [comment]
@IsaSih This is not wired in everywhere, only the growls (coloured text modals that hover over page content close to the user avatar) and the 'welcome to rancher' on upgrade message. 

If no growl was shown before (upgrade message aside), there is no 2.12.0 expectation to show the notification

**Comment by IsaSih** [comment]
I'm closing this ticket as complete then. But are there any issues to track adding growl to these scenarios? (ex> installing an extension, provisioning a cluster, etc)

**Comment by IsaSih** [comment]
Found this https://github.com/rancher/dashboard/issues/11791
